parcels/field.py

```diff
@@ -39,7 +39,11 @@ def __init__(self, name, data, lon, lat, depth=None, time=None,
             # Make a copy of the transposed array to enforce
             # C-contiguous memory layout for JIT mode.
             self.data = np.transpose(self.data).copy()
-        self.data = self.data.reshape((self.time.size, self.lat.size, self.lon.size))
+        if self.depth.size > 1:
```

Comment:
Having two potential shapes for `self.data` creates a whole lot of special casing that can be avoided by defaulting to a `np.zeros(1, dtype=np.float32)` in the constructor above. This should be treated the same way we treat time and it will also make things easier when dealing with the interpolators below, I think.